src/Transform/BundleMemoryPools.cpp

```diff
@@ -160,20 +161,34 @@ class KrnlBundleStaticMemoryPools : public OpRewritePattern<AllocOp> {
     // Get parent block.
     Block *parentBlock = allocOp.getOperation()->getBlock();
 
+    // Check if parent block has been seen before.
     if (blockToStaticPool->count(parentBlock) == 0) {
       allocOp.getOperation()->moveBefore(&parentBlock->front());
       // Create new entry in the block map.
-      blockToStaticPool->insert(
-          std::pair<Block *, AllocOp>(parentBlock, allocOp));
+      AlignmentToMemPool *alignmentToMemPool = new AlignmentToMemPool();
+      blockToStaticPool->insert(std::pair<Block *, AlignmentToMemPool *>(
+          parentBlock, alignmentToMemPool));
+    }
+
+    // Populate alignment integer.
+    int64_t alignment = getAllocAlignment(allocOp);
 
-      // This is the initial memory pool for this block and it is
-      // trivially bundled hence it's safe to return success.
+    // If this parent block has been found present in the map, check that
+    // a static memory bundle with the current alignment already exists.
+    AlignmentToMemPool *alignmentToMemPool = blockToStaticPool->at(parentBlock);
+    if (alignmentToMemPool->count(alignment) == 0) {
+      // If static memory bundle for this alignment does not exist, then
+      // create an entry.
+      alignmentToMemPool->insert(
+          std::pair<int64_t, AllocOp>(alignment, allocOp));
+
+      // This is the initial memory pool for this block and alignment
+      // so trivially bundle it and return success.
       return success();
     }
 
-    // If this parent block has been found present in the map, it means
-    // a static memory bundle already exists. Fetch it.
-    AllocOp staticMemPoolAlloc = blockToStaticPool->at(parentBlock);
+    // Static memory pool for this alignment exists, fetch it.
+    AllocOp staticMemPoolAlloc = alignmentToMemPool->at(alignment);
 
     // If this is the alloc representing the memory pool and the function
     // already has an init block, pattern matching must fail to avoid
```

Comment:
Avoid what?

Reply:
the answer is on the next line :) (although it should say static memory pool not dynamic)